addon/mixins/ripple-mixin.js

```diff
@@ -1,287 +1,286 @@
-      }, (this.get('outline') ? 450 : 225));
-    }, 0);
-
-    return elem;
+    }
+  },
+  handleMouseup() {
+    this.autoCleanup(this, this.clearRipples);
   },
-  removeElement(elem, wait) {
-    var ripples = this.ripples;
-    ripples.splice(ripples.indexOf(elem), 1);
-    if (ripples.length === 0 && this.rippleContainer) {
-      this.rippleContainer.css({ backgroundColor: '' });
+  handleTouchmove() {
+    this.autoCleanup(this, this.deleteRipples);
+  },
+  deleteRipples() {
+    for (let i = 0; i < this.ripples.length; i++) {
+      this.ripples[i].remove();
     }
-    Ember.run.later(this,function(){
-      elem.remove();
-    }, wait);
   },
-  updateElement(elem) {
-    var index = this.ripples.indexOf(elem),
-    state = this.rippleStates[index] || {},
-    elemIsActive = this.ripples.length > 1 ? false : this.isActive,
-    elemIsHeld   = this.ripples.length > 1 ? false : this.isHeld;
-    if (elemIsActive || state.animating || elemIsHeld) {
-      elem.addClass('md-ripple-visible');
-    } else if (elem) {
-      elem.removeClass('md-ripple-visible');
-      if (this.get('outline')) {
-        elem.css({
-          width: this.rippleSize + 'px',
-          height: this.rippleSize + 'px',
-          marginLeft: (this.rippleSize * -1) + 'px',
-          marginTop: (this.rippleSize * -1) + 'px'
-        });
-      }
-      this.removeElement(elem, this.get('outline') ? 450 : 650);
+  clearRipples() {
+    for (let i = 0; i < this.ripples.length; i++) {
+      this.fadeInComplete(this.ripples[i]);
     }
   },
-  /**
-  * Generates the ripple css
-  *
-  * @param {number} the diameter of the ripple
-  * @param {number} the left cursor offset
-  * @param {number} the top cursor offset
-  *
-  * @returns {{backgroundColor: *, width: string, height: string, marginLeft: string, marginTop: string}}
-  */
-  getRippleCss(size, left, top) {
-    var rect,
-    css = {
-      backgroundColor: rgbaToRGB(this.color),
-      borderColor: rgbaToRGB(this.color),
-      width: size + 'px',
-      height: size + 'px'
-    };
+  createContainer() {
+    let container = Ember.$('<div class="md-ripple-container"></div>');
+    this.rippleElement.append(container);
+    return container;
+  },
+  clearTimeout() {
+    if (this.timeout) {
+      Ember.run.cancel(this.timeout);
+      this.timeout = null;
+    }
+  },
+  isRippleAllowed() {
+    let element = this.rippleElement[0];
+    do {
+      if (!element.tagName || element.tagName === 'BODY') {
+        break;
+      }
 
-    if (this.get('outline')) {
-      css.width = 0;
-      css.height = 0;
-    } else {
-      css.marginLeft = css.marginTop = (size * -0.5) + 'px';
+      if (element && Ember.$.isFunction(element.hasAttribute)) {
+        if (element.hasAttribute('disabled')) {
+          return false;
+        }
+        if (this.inkRipple() === false) {
+          return false;
+        }
+      }
+
+    } while (element = element.parentNode);
+
+
+    return true;
+  },
+  inkRipple() {
```

Comment:
Since `inkRipple()` is basically a wrapper around `rippleInk` computed, we can replace all `inkRipple()` entries with `this.get('rippleInk')`.